source/userguide/text_preparation_pipeline.rst

```diff
@@ -0,0 +1,661 @@
+based on the language of the application.
+
+
+EnglishNLTKStemmer
+^^^^^^^^^^^^^^^^^^
+
+The :attr:`EnglishNLTKStemmer` stemmer uses the :attr:`PorterStemmer` from the nltk library. The porter stemmer implements a series of rules that determine common suffixes from sentences.
+This includes removing the letter "s" from plural words or "ing" from gerunds.
+
+Let's take a look at a few examples of the :attr:`EnglishNLTKStemmer`. First we'll make an instance of the stemmer:
+
+.. code:: python
+
+    from mindmeld.text_preparation.stemmers import EnglishNLTKStemmer
+    english_nltk_stemmer = EnglishNLTKStemmer()
+
+Now let's stem the words "running" and "pearls".
+
+.. code:: python
+
+    >>> print(english_nltk_stemmer.stem_word("running"))
+    >>> run
+    >>> print(english_nltk_stemmer.stem_word("pearls"))
+    >>> pearl
+
+As expected, the stemmer removes "ing" from "running" and the "s" from "pearls" to create stemmed words.
+
+
+SnowballNLTKStemmer
+^^^^^^^^^^^^^^^^^^^
+
+The :attr:`SnowballNLTKStemmer` stemmer works in a similar manner to the :attr:`EnglishNLTKStemmer`, however, it offers support for a larger set of languages.
+Namely, the :attr:`SnowballNLTKStemmer` supports Danish (da), Dutch (nl), Finnish (fi), French (fr), German (de), Hungarian (hu), Italian (it), Norwegian (nb), Portuguese (pt), Romanian (ro), Russian (ru), Spanish (es) and Swedish (sv).
+
+To create an instance of the :attr:`SnowballNLTKStemmer`, we can use MindMeld's :attr:`StemmerFactory`.
+
+.. code:: python
+
+    from mindmeld.text_preparation.stemmers import StemmerFactory
+    es_snowball_stemmer = StemmerFactory.get_stemmer_by_language("es")
+
+Now let's stem the words "corriendo" ("running") and "perlas" ("pearls").
+
+.. code:: python
+
+    >>> print(english_nltk_stemmer.stem_word("corriendo"))
+    >>> corr
+    >>> print(english_nltk_stemmer.stem_word("perlas"))
+    >>> perl
+
+As expected, the stemmer removes "iendo" from "corriendo" and the "as" from "perlas" to create stemmed words.
+
+
+Creating a Custom Stemmer
+^^^^^^^^^^^^^^^^^^^^^^^^^
+This section includes boilerplate code to build a custom stemmer class.
+A custom stemmer must extend from MindMeld's abstract :attr:`Stemmer` class:
+
+
+.. code:: python
+
+    from abc import ABC, abstractmethod
+
+    class Stemmer(ABC):
+
+        @abstractmethod
+        def stem_word(self, word):
+            """
+            Gets the stem of a word. For example, the stem of the word 'fishing' is 'fish'.
+
+            Args:
+                word (str): The word to stem
+
+            Returns:
+                stemmed_word (str): A stemmed version of the word
+            """
+            raise NotImplementedError
+
+Let's create a stemmer that only removes the "-ing" suffix if found at the end of a word. We'll call it the :attr:`GerundSuffixStemmer`.
```

Comment:
A more plausible example would be more helpful--would it be to much to show how to use spacy lexemes as stems?